ocs_ci/utility/kms.py

```diff
@@ -0,0 +1,503 @@
+            client_cert_data = templating.load_yaml(
+                constants.EXTERNAL_VAULT_CLIENT_CERT
+            )
+            self.client_cert_name = config.ENV_DATA.get(
+                "VAULT_CLIENT_CERT", constants.VAULT_DEFAULT_CLIENT_CERT
+            )
+            client_cert_data["metadata"]["name"] = self.client_cert_name
+            client_cert_data["data"]["cert"] = self.client_cert_base64
+            self.create_resource(client_cert_data, prefix="clientcert")
+
+            # create client key secert
+            client_key_data = templating.load_yaml(constants.EXTERNAL_VAULT_CLIENT_KEY)
+            self.client_key_name = config.ENV_DATA.get(
+                "VAULT_CLIENT_KEY", constants.VAULT_DEFAULT_CLIENT_KEY
+            )
+            self.client_key_name["metadata"]["name"] = self.client_key_name
+            client_key_data["data"]["key"] = self.client_key_base64
+            self.create_resource(client_key_data, prefix="clientkey")
+
+        # create oc resource secret for token
+        token_data = templating.load_yaml(constants.EXTERNAL_VAULT_KMS_TOKEN)
+        # token has to base64 encoded (with padding)
+        token_data["data"]["token"] = base64.b64encode(
+            # encode() because b64encode expects a byte type
+            self.vault_path_token.encode()
+        ).decode()  # decode() because b64encode returns a byte type
+        self.create_resource(token_data, prefix="token")
+
+        # create ocs-kms-connection-details
+        connection_data = templating.load_yaml(
+            constants.EXTERNAL_VAULT_KMS_CONNECTION_DETAILS
+        )
+        connection_data["data"][
+            "VAULT_ADDR"
+        ] = f"https://{self.vault_server}:{self.port}"
+        connection_data["data"]["VAULT_BACKEND_PATH"] = self.vault_backend_path
+        connection_data["data"]["VAULT_CACERT"] = self.ca_cert_name
+        connection_data["data"]["VAULT_CLIENT_CERT"] = self.client_cert_name
+        connection_data["data"]["VAULT_CLIENT_KEY"] = self.client_key_name
+        self.vault_namespace = config.ENV_DATA.get(
+            "VAULT_NAMESPACE", constants.VAULT_DEFAULT_NAMESPACE
+        )
+        connection_data["data"]["VAULT_NAMESPACE"] = self.vault_namespace
+        connection_data["data"]["VAULT_TLS_SERVER_NAME"] = self.vault_tls_server
+        self.create_resource(connection_data, prefix="kmsconnection")
+
+    def create_resource(self, resource_data, prefix=None):
+        """
+        Given a dictionary of resource data, this function will
+        creates oc resource
+
+        Args:
+            resource_data (dict): yaml dictionary for resource
+            prefix (str): prefix for NamedTemporaryFile
+
+        """
+        resource_data_yaml = tempfile.NamedTemporaryFile(
+            mode="w+", prefix=prefix, delete=False
+        )
+        templating.dump_data_to_temp_yaml(resource_data, resource_data_yaml.name)
+        run_cmd(f"oc create -f {resource_data_yaml.name}", timeout=1200)
```

Comment:
Do we need timeout of 1200 for create command?

Reply:
copy paste mistake, reducing it to 300

Reply:
done